### PR TITLE
Fix gradient checkpointing bug in Speecht5

### PR DESCRIPTION
**Pull request by KMFODA**
This PR fixes a bug that a user can encounter while using generate and models that use gradient_checkpointing.

Fixes Issue https://github.com/huggingface/transformers/issues/21737

cc @younesbelkada or @gante

### CODE REVIEW

**Review by younesbelkada — COMMENTED** [review]
Thanks a lot for your PR! Can you use `warning_once` instead?

**Comment by KMFODA** [comment]
Sorry fixed that

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for this!

File: src/transformers/models/speech_to_text/modeling_speech_to_text.py [hunk 1]
```diff
@@ -1024,6 +1024,13 @@ def forward(
         hidden_states = inputs_embeds + positions
         hidden_states = nn.functional.dropout(hidden_states, p=self.dropout, training=self.training)
 
+        if self.gradient_checkpointing and self.training:
+            if use_cache:
+                logger.warning_once(
+                    "`use_cache = True` is incompatible with gradient checkpointing. Setting `use_cache =" " False`..."
```
Comment (gante) [inline src/transformers/models/speech_to_text/modeling_speech_to_text.py:1030]:
```suggestion
                    "`use_cache = True` is incompatible with gradient checkpointing. Setting `use_cache = False`..."
```

**Review by gante — APPROVED** [review]
Thank you for the contribution 🙏  (took the liberty to do a minor string surgery before merging, I hope you don't mind!)

**Comment by gante** [comment]
@KMFODA it is possible that this is the wrong branch -- the diff doesn't contain changes to speecht5 :)

**Comment by younesbelkada** [comment]
Yes the diff seems to have changed a bit, can you please double check 🙏 ?

**Comment by KMFODA** [comment]
Sorry yes I think the branches were mixed up. This should now introduce the changes to SpeechT5 and fix the formatting issues @gante spotted in [modeling_speech_to_text.py](https://github.com/huggingface/transformers/pull/22080/commits/4da5f4baaf4eddb71787ba4dbbbd3791953e481a).

**Comment by gante** [comment]
Perfect, thank you for the changes @KMFODA 💛
